### PR TITLE
SEP-10 3.0

### PR DESCRIPTION
**Pull request by Kirbyrawr**
Added homeDomain check in WebAuthentication.cs

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [X] I have read the **CONTRIBUTING** document.
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Review by leighmcculloch — APPROVED** [review]
🎉 I think this looks great. I left one suggestion (💡) that I don't think is required to support SEP-10 v3.0.0, but may be relevant depending on the use cases the SDK wishes to support.

File: stellar-dotnet-sdk/WebAuthentication.cs [hunk 1]
```diff
@@ -145,6 +145,9 @@ public static class WebAuthentication
             if (operation.SourceAccount is null)
                 throw new InvalidWebAuthenticationException("Challenge transaction operation must have source account");
 
+            if (operation.Name.Split(' ')[0] != homeDomain)
```
[the remark this answers is not in the document]
Comment (Kirbyrawr) [reply stellar-dotnet-sdk/WebAuthentication.cs]: Hi, this is fixed now!

[the remark this answers is not in the document]
Comment (leighmcculloch) [reply stellar-dotnet-sdk/WebAuthentication.cs]: Nice! Looks great!

**Review by fracek — APPROVED** [review]
Looks good to me. Thanks for this.

**Pull request merged**